sway-core/src/transform/to_parsed_lang/context.rs

```diff
@@ -0,0 +1,66 @@
+use core::cell::RefCell;
+
+/// A context containing global information used during `convert_parse_tree"
+#[derive(Default)]
+pub struct Context {
+    /// The inner context.
+    /// This construction is used to avoid `&mut` all over `convert_parse_tree`.
+    inner: RefCell<ContextInner>,
+}
```

Comment:
Using `RefCell` gives me the heebie-jeebies as they can accidentally produce _runtime_ borrow conflicts.  It looks below like it's used pretty safely, but I'm not sure I follow the justification of avoiding passing `&mut`?  What's wrong with a mutable context?

Reply:
Perhaps we could also keep a `&Handler` in the `Context` for the traversal? Could always be done in a follow-up PR though too.

Reply:
To be entirely honest, `Context` here is modelled after the [`Handler`](https://github.com/FuelLabs/sway/blob/master/sway-error/src/handler.rs) which uses `RefCell`. I don't really mind one way over another.. just went with `RefCell` for consistency. Thoughts? 

@mitchmindtree moving `Handler` into `Context` in this file seems like a good idea!

Reply:
> To be entirely honest, `Context` here is modelled after the [`Handler`](https://github.com/FuelLabs/sway/blob/master/sway-error/src/handler.rs) which uses `RefCell`. I don't really mind one way over another.. just went with `RefCell` for consistency. Thoughts?

I see.  Personally I'd rather avoid them unless there's no (sane) alternative.

Reply:
Alright, I'm no longer using `RefCell` for `Context`. I can update `Handler` in a separate PR along with moving `Handler` into `Context`.